### DISCUSSION TITLE
Data logging or automation for vintage roaster?

### DISCUSSION THREAD

**Discussion by georgemvg** [artisan-roaster-scope/artisan#1068] [Q&A]
Hi all!
I bought a very old roaster this year, and I am trying to find out what more I can do besides connecting the roaster's thermocouples to artisan.
Lets say that I find a way to control from artisan the gas pressure via a proportonal valve and an output phidget, and the 3phase ac fan with a pwm module. Will artisan try to replicate the given artisan graph?
I do not wish yet to use plc and ladder to program the operation of the roaster.
I wish also to be able to use it manually, as I do not want to ruin the "old" operation, like the vertical gas glass pressure gauge.
Can artisan use a physical slider/knob as input and then artisan to give the output to valve/pwm for the changes to be made to the roaster?
In that case, if a proportional valve will be used, I am thinking about leaving full open the gas regulator until it has to be used manually.
Excuse my ignorance.
Just trying to fully understand how artisan works.

**Comment by EspressoMachineWhisperer** [comment]
Hi. Answer is yes, Artisan can be used in place of a PLC. We use Artisan to control IR Heaters, Fan speed, drum speed etc. It can drive the proportional valve for your Gas. We use Phidgets to do all the Outputs and Inputs, both analog and digital. Unfortunately Phidgets don't conform to standard instrumentation signal formats so there is some requirement for conversion, but we simply do that using span-able signal isolators.
